eng/targets/Settings.props

```diff
@@ -195,7 +195,7 @@
     <When Condition="'$(Language)' == 'C#'">
       <PropertyGroup>
         <LangVersion>preview</LangVersion>
-        <WarningLevel>4</WarningLevel>
+        <WarningLevel>9999</WarningLevel>
```

Comment:
Wasn't this supposed to be [`<AnalysisLevel>latest</AnalysisLevel>`](
 https://devblogs.microsoft.com/dotnet/automatically-find-latent-bugs-in-your-code-with-net-5/)?

Reply:
@alrz From what I understand, AnalysisLevel is related to the shipped analyzers (CAxxxx) in .NET SDK, not the compiler diagnostics. See https://docs.microsoft.com/dotnet/fundamentals/code-analysis/overview#latest-updates

Reply:
perhaps we should adopt AnalysisLevel instead anyway..?

Reply:
it is about analyzers but from the article it does affect warning level as well.

Reply:
I couldn't see where in the article it mentions warning waves, but if it does affect it, then definitely go with it. If not, then probably add both.

Reply:
I recommend setting this to `<AnalysisLevel>preview</AnalysisLevel>` only if we also want the additional analyzers to be brought along for the ride. As roslyn has rather complex analyzer consumption requirements (as we both build and consume the same analyzers in our build) I recommend `<WarningLevel>9999</WarningLevel>`